### PR TITLE
Devex 851: case helper, hoisting correspondence and document updates

### PR DESCRIPTION
**Pull request by kkoskelin**
also: using existing persistence methods instead of duplicated functions that directly talk to the document client.

### CODE REVIEW

File: shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.js [hunk 1]
```diff
@@ -1,5 +1,100 @@
...
+/**
+ * Identifies documents which have been updated and issues persistence calls
+ *
+ * @param {object} args the arguments for updating the case
+ * @param {object} args.applicationContext the application context
+ * @param {object} args.caseToUpdate the case with its updated document data
+ * @param {object} args.oldCase the case as it is currently stored in persistence, prior to these changes
+ * @returns {Array<Promise>} the persistence request promises
+ */
+const updateCaseDocuments = ({ applicationContext, caseToUpdate, oldCase }) => {
```
Comment (rachaelparris) [inline shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.js]: Nit: could this be renamed `updateCaseDocketEntries` for clarity?

**Review by matthopson — APPROVED** [review]
nod

File: shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.js (hunk 1 above)
Comment (matthopson) [reply shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.js]: Ohhh, good call.

File: shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.test.js [hunk 2]
```diff
@@ -131,4 +157,124 @@ describe('updateCaseAndAssociations', () => {
...
+      ).toMatchObject({ applicationContext, caseToUpdate, oldCase });
+
+      expect(
+        applicationContext.getPersistenceGateway().updateDocketEntry,
+      ).toHaveBeenCalledTimes(4);
+    });
+  });
+
+  describe('correspondences', () => {
+    it('does not call updateCaseCorrespondence if all docket entries are unchanged', async () => {
```
Comment (rachaelparris) [inline shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.test.js]: Nit: test description `docket entries` -> `correspondences` ?

File: shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.test.js [hunk 3]
```diff
@@ -131,4 +157,124 @@ describe('updateCaseAndAssociations', () => {
...
+      await updateCaseAndAssociations({
+        applicationContext,
+        caseToUpdate: validMockCase,
+      });
+      expect(
+        applicationContext.getPersistenceGateway().updateCaseCorrespondence,
+      ).not.toHaveBeenCalled();
+    });
+
+    it('calls updateCaseCorrespondence for each case document which has been added or changed', async () => {
```
Comment (rachaelparris) [inline shared/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.test.js]: Nit: `document` in test description -> `correspondence`

**Review by rachaelparris — APPROVED** [review]
I see that you just moved some things around and some of my comments are not on your original code, but probably a good time to fix them anyway! 😄

**Review by rachelschneiderman — APPROVED** [review]

**Pull request merged**